deploy/runtools/firesim_topology_elements.py

```diff
@@ -293,27 +288,66 @@ def copy_back_job_results_from_run(self, slotno):
                     run("""sudo chattr -i {}/etc/sysconfig/nfs""".format(mountpoint))
                 run("""sudo chown -R centos {}""".format(mountpoint))
 
-            ## copy back files from inside the rootfs
-            with warn_only(), StreamLogger('stdout'), StreamLogger('stderr'):
+        if not self.get_job().parent_workload.compress_results:
```

Comment:
Please reduce the code duplication across both sides of the conditional. Perhaps it makes sense to always compress before transporting. Alternatively, build up a filelist first before either compressing or iterating through them with gets, and then do the common clean up tasks.